apps/examples/mediaplayer/mediaplayer_main.cpp

```diff
@@ -267,16 +296,32 @@ void MyMediaPlayer::onPlaybackPaused(MediaPlayer &mediaPlayer)
 	cout << "onPlaybackPaused" << endl;
 }
 
-void MyMediaPlayer::onFocusChange(int focusChange)
+void MyMediaPlayer::onAsyncPrepared(MediaPlayer &mediaPlayer, player_error_t error)
 {
+	cout << "onAsyncPrepared res" << endl;
+	if (error == PLAYER_ERROR_NONE) {
+		mediaPlayer.start();
+	} else {
+		mediaPlayer.unprepare();
+	}
+}
 
+void MyMediaPlayer::onFocusChange(int focusChange)
+{
 	switch (focusChange) {
 	case FOCUS_GAIN:
-		if (mp.prepare() != PLAYER_OK) {
-			cout << "Mediaplayer::prepare failed" << endl;
-		}
-		if (mp.start() != PLAYER_OK) {
-			cout << "Mediaplayer::start failed" << endl;
+		if (testSource == TEST_HTTP) {
+			if (mp.prepareAsync() != PLAYER_OK) {
+				cout << "Mediaplayer::prepareAsync failed" << endl;
+			}
+			// start player in onAsyncPrepared()
+		} else {
+			if (mp.prepare() != PLAYER_OK) {
+				cout << "Mediaplayer::prepare failed" << endl;
```

Comment:
I have a question, if mp prepare failed, does it also may start succefully? 
I mean do we return when mp prepare failed?

Reply:
If prepare player fail, player state is not READY, then start will also fail.
I think it's ok to give error log in case of failure without return.